pkgs/build-support/cc-wrapper/add-hardening.sh

```diff
@@ -1,33 +1,45 @@
-hardeningFlags=(fortify stackprotector pic strictoverflow format relro bindnow)
-# Intentionally word-split in case 'hardeningEnable' is defined in
-# Nix. Also, our bootstrap tools version of bash is old enough that
-# undefined arrays trip `set -u`.
-if [[ -v hardeningEnable[@] ]]; then
-  hardeningFlags+=(${hardeningEnable[@]})
-fi
+allHardeningFlags=(fortify stackprotector pie pic strictoverflow format relro bindnow)
 hardeningCFlags=()
 
-declare -A hardeningDisableMap
+declare -A hardeningEnableMap=()
 
-# Intentionally word-split in case 'hardeningDisable' is defined in Nix.
-for flag in ${hardeningDisable[@]:-IGNORED_KEY} @hardening_unsupported_flags@
-do
-  hardeningDisableMap[$flag]=1
+# Intentionally word-split in case 'NIX_HARDENING_ENABLE' is defined in Nix. The
+# array expansion also prevents undefined variables from causing trouble with
+# `set -u`.
+for flag in ${NIX_HARDENING_ENABLE-}; do
```

Comment:
`s/NIX_/NIX_@infixSalt@_/`

Reply:
And the the corresponding `add-flags.sh` change I mentioned before.

Reply:
> `s/NIX_/NIX_@infixSalt@_/`

Working on that now.

> And the the corresponding add-flags.sh change I mentioned before.

Oh, I modified `pkgs/build-support/bintools-wrapper/add-flags.sh`. Do I want to modify both add-flags? I guess, with the split vars approach, I'd want to put the respective var name in each.